### PR TITLE
front end validation

### PR DESCRIPTION
**Pull request by cookiedan42**
## Overview
#78 

Add front end validation for username and password

### Main changes and their rationale(s)

* followed style of text field validation error used in signup/login 

Some considerations were y

imported the same checker from login, so we can set stricter rules and standardize

### Things to take note of

* user-service response was not standardized, so front end wasn't picking it up
* used existing snackbar for error message
* tldr snackbar for back end errors, text box for front end validation error

### Testing

e.g. Simply run:
```
cypress <some file>.ts
```

image

### CODE REVIEW

**Comment by cookiedan42** [comment]
image
image

File: user-service/src/constants.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ export const ENV_IS_PROD = process.env.ENV == 'PROD'
  */
 export const CORS_OPTIONS: cors.CorsOptions = {
   credentials: true,
+  // origin: '*'
```
Comment (tanyjnaaman) [inline user-service/src/constants.ts:30]: Remove this?

**Review by tanyjnaaman — APPROVED** [review]
